test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
                         "requested hbmem of %d mb" %
                         (hbm_assig, v_name, hbm_req))
                     self.assertTrue(False)
+
+        # Delete the Job.
+        self.server.delete(job_id, wait=True)
+
+    def test_job_request_insufficent_hbmemm_rsc(self):
+        """
+        Submit a job request that requests more than available HBMEM.
+        Check if the job is in the 'Q' state with valid comment.
+        Delete the job
+        """
+
+        knl_vnodes = self.get_knl_vnodes()
+
+        if len(knl_vnodes) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (knl_vnodes))
+
+        hbm_req = 18000
+        a = {'Resource_List.select': '1:hbmem=%dmb' % hbm_req}
+        job = Job(TEST_USER, attrs=a)
+
+        job_id = self.server.submit(job)
+
+        # Check that job is in Q state with valid comment
+        job_comment = "Not Running: Insufficient amount of resource: hbmem"
+        self.server.expect(JOB, {'job_state': 'Q', 'comment':
+                           (MATCH_RE, job_comment)}, id=job_id)
+
         # Delete the Job.
         self.server.delete(job_id, wait=True)
+
+    def test_job_request_knl(self):
+        """
+        Create a job that requests aoe should run on a KNL vnode.
+        Submit the job to the Server.Check if the job runs on a KNL vnode
+        and if the job is in the 'R' state.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='Test is NA for Craysim')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set provisioning attributes on KNL vnode.
+        self.set_provisioning()
+
+        # Submit job that request aoe
+        req_aoe = self.request_current_aoe()
+        job = Job(TEST_USER)
+
+        job.create_script(
+                          "#PBS -joe -o localhost:/tmp -lselect=1:ncpus=1:aoe=%s\n" % req_aoe +
```

Comment:
Consider using a triple quoted string for the job script.  If you want to continue to use this string, merge the above two lines together.

Reply:
Since I am synthesizing the script using a variable req_aoe, i will not be able to use triple quotes. Also If I merge two lines I would surpass 80 characters limit